Core/Core/AudioVideo/AudioPlayerViewController.swift

```diff
@@ -0,0 +1,183 @@
+
+    lazy var formatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        formatter.allowedUnits = [.minute, .second]
+        formatter.unitsStyle = .positional
+        formatter.zeroFormattingBehavior = .pad
+        return formatter
+    }()
+    var player: AVAudioPlayer?
+    var timer: CADisplayLink?
+
+    public static func create() -> AudioPlayerViewController {
+        let controller = loadFromStoryboard()
+        controller.loadViewIfNeeded() // So the didSets work correctly.
+        return controller
+    }
+
+    public var accessibilityPrefix: String = "AudioPlayer." {
+        didSet {
+            currentTimeLabel?.accessibilityIdentifier = currentTimeLabel?.accessibilityIdentifier?.replacingOccurrences(of: oldValue, with: accessibilityPrefix)
+            loadingView?.accessibilityIdentifier = loadingView?.accessibilityIdentifier?.replacingOccurrences(of: oldValue, with: accessibilityPrefix)
+            playPauseButton?.accessibilityIdentifier = playPauseButton?.accessibilityIdentifier?.replacingOccurrences(of: oldValue, with: accessibilityPrefix)
+            remainingTimeLabel?.accessibilityIdentifier = remainingTimeLabel?.accessibilityIdentifier?.replacingOccurrences(of: oldValue, with: accessibilityPrefix)
+            timeSlider?.accessibilityIdentifier = timeSlider?.accessibilityIdentifier?.replacingOccurrences(of: oldValue, with: accessibilityPrefix)
+        }
+    }
+
+    public var backgroundColor: UIColor? {
+        get { return backgroundView?.backgroundColor }
+        set { backgroundView?.backgroundColor = newValue }
+    }
+
+    public var color: UIColor = .white {
+        didSet {
+            currentTimeLabel?.textColor = color
+            loadingView?.color = color
+            playPauseButton?.tintColor = color
+            remainingTimeLabel?.textColor = color
+            thumb?.backgroundColor = color
+            track?.backgroundColor = color.withAlphaComponent(0.5)
+            trackFill?.backgroundColor = color.withAlphaComponent(0.5)
+        }
+    }
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        backgroundColor = .named(.backgroundDarkest)
+        playPauseButton?.accessibilityLabel = NSLocalizedString("Play", bundle: .core, comment: "")
+        currentTimeLabel?.accessibilityLabel = NSLocalizedString("Time elapsed", bundle: .core, comment: "")
+        loadingView?.accessibilityIdentifier = "AudioPlayer.loadingView"
+        loadingView?.color = color
+        remainingTimeLabel?.accessibilityLabel = NSLocalizedString("Total time", bundle: .core, comment: "")
+        timeSlider?.accessibilityLabel = NSLocalizedString("Current position", bundle: .core, comment: "")
+    }
+
+    public func load(url: URL?) {
+        currentTimeLabel?.text = NSLocalizedString("--:--", bundle: .core, comment: "")
+        remainingTimeLabel?.text = NSLocalizedString("--:--", bundle: .core, comment: "")
```

Comment:
I wonder if these strings should have a comment so that the translators know this is a timestamp empty state?

Reply:
yeah, I should add that.